app/views/products/index.html.erb

```diff
@@ -0,0 +1,20 @@
+<h1>All Products</h1>
+
+<table>
+  <tr>
+    <th>Title</th>
+    <th>Description</th>
+    <th>Status</th>
+    <th>Vendor</th>
+    <th>Average price</th>
+  </tr>
+  <% @products.each do |product| %>
+    <tr>
+      <td><%= product.title %></td>
+      <td><%= product.body_html %></td>
+      <td><%= product.status %></td>
+      <td><%= product.vendor %></td>
+      <td><%= product.product_variants.average(:price).to_f %></td>
```

Comment:
Please, comment on this moment: shall I transfer average prices to model? Each product has many variants so I found it difficult to initially display prices of all variants

Reply:
This is potential N+1 problem. You can leave as is for now, but add preload.

Reply:
Done!